example/main.rb

```diff
@@ -2,11 +2,11 @@
 require_relative 'gen/haberdasher_pb.rb'
 require_relative 'gen/haberdasher_twirp.rb'
 
-class HaberdasherImplementation
-    def HelloWorld(req)
-        return Examples::HelloWorldResponse.new(message: "Hello #{req.name}")
+class HaberdasherHandler
+    def hello_world(req)
+        return Example::HelloWorldResponse.new(message: "Hello #{req.name}")
     end
```

Comment:
could tide the example a little more:
```ruby
class HaberdasherHandler
  def hello_world(req)
    return {message: "Hello #{req.name}"}
  end
end
```

Don't worry about using the right "type", because the `Example::HelloWorldResponse` class is already being used by the Service routing, so the error message you would get if you use a wrong property is exactly the same (the Hash is used as argument in the constructor, which validates the keys passed).